### PR TITLE
Dlmoreram

### PR DESCRIPTION
**Pull request by wsand02**
This PR cleans up and improves a lot of things. The biggest change is a full rewrite of the audio pipeline: WAV decoding and MP3 encoding are now merged into a single module with proper error handling and tests for different formats and edge cases. The upload API has been updated to use this new pipeline, saving files to disk and returning encoded MP3s with clearer error responses.

On the API side, error/status structs are now public and better covered by tests, including authentication checks.

Dependencies have been updated (uuid, job_scheduler_ng, tempfile), the crate version bumped to 1.1.1, and the README now includes setup details for the data dir/config.

Docs and CI got some polish too: badges were added to the README, and a new GitHub Actions job runs Clippy to enforce linting.

### CODE REVIEW

File: src/audio.rs [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+            right.resize(max_len, 0.0);
+            encode_dual(&left, &right, &mut bwriter, &mut mp3_encoder)?;
+            left.clear();
+            right.clear();
+        }
+    } else if !left.is_empty() {
+        encode_mono(&left, &mut bwriter, &mut mp3_encoder)?;
+        left.clear();
+    }
+    let num_frames = cmp::max(left.len(), right.len());
```
[the remark this answers is not in the document]
Comment (wsand02) [reply src/audio.rs:138]: Num frames is checked before final write / flush.... 😡 (I am talking to the wall).
